src/ng-xform/field-components/base-dynamic-field.component.ts

```diff
@@ -11,9 +11,19 @@ export class BaseDynamicFieldComponent<T extends DynamicField> implements OnInit
   @Input() editing = true;
 
   control: FormControl;
+  visible = true;
 
   ngOnInit() {
     this.control = <FormControl>this.form.controls[this.field.key];
+    if (this.field.visibilityFn) {
+      this.form.valueChanges.subscribe(val => {
```

Comment:
Don't you need to unsubscribe on the component destruction?

Reply:
you're right.